app/src/main/java/com/alexbar10/cryptotrack/ui/cryptoAvailable/CryptocurrenciesAvailableFragment.kt

```diff
@@ -0,0 +1,107 @@
+package com.alexbar10.cryptotrack.ui.cryptoAvailable
+
+import android.net.ConnectivityManager
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.core.view.isVisible
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.Observer
+import androidx.navigation.NavController
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.GridLayoutManager
+import com.alexbar10.cryptotrack.MainActivity
+import com.alexbar10.cryptotrack.R
+import com.alexbar10.cryptotrack.databinding.FragmentCryptocurrenciesAvailableBinding
+import com.alexbar10.cryptotrack.domain.Cryptocurrency
+import com.alexbar10.cryptotrack.networking.NetworkStatusChecker
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+class CryptocurrenciesAvailableFragment : Fragment() {
+
+    private var _binding: FragmentCryptocurrenciesAvailableBinding? = null
+    private val binding: FragmentCryptocurrenciesAvailableBinding
+        get() = _binding!!
+    private lateinit var navController: NavController
+    private val viewModel: CryptocurrenciesAvailableViewModel by viewModels()
+    private lateinit var cryptocurrenciesAdapter: CryptocurrenciesAdapter
+    private val networkStatusChecker by lazy {
+        NetworkStatusChecker(requireActivity().getSystemService(ConnectivityManager::class.java))
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
+        return FragmentCryptocurrenciesAvailableBinding
+            .inflate(layoutInflater, container, false)
+            .apply { _binding = this }
+            .root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        cryptocurrenciesAdapter = CryptocurrenciesAdapter { cryptocurrencySelected ->
+            CryptocurrenciesAvailableFragmentDirections
+                .actionCryptocurrenciesAvailableFragmentToCryptocurrencyDetailsFragment(cryptocurrencySelected)
+                .let { navController.navigate(it) }
+        }
+        navController = findNavController()
+        (requireActivity() as MainActivity).supportActionBar?.title = getString(R.string.app_name)
+
+        binding.cryptocurrenciesRecyclerView.run {
+            adapter = cryptocurrenciesAdapter
+            layoutManager = GridLayoutManager(context, 2)
+        }
+        binding.marketMxnCheckbox.setOnCheckedChangeListener { _, isChecked -> viewModel.showCryptocurrenciesFor("mxn", isChecked) }
```

Comment:
You can probably define a Class to define all these Currencies (mxn, ars, btc, etc) as constants, that way we don't hard-code the strings in the code and we can reuse them in other places (as needed).